client/my-sites/themes/theme-showcase.jsx

```diff
@@ -137,7 +116,7 @@ class ThemeShowcase extends Component {
 		}
 	}
 
-	isStaticFilter = ( tabFilter ) => Object.values( this.staticFilters ).includes( tabFilter );
+	isStaticFilter = ( tabFilter ) => Object.values( this.getStaticFilters() ).includes( tabFilter );
```

Comment:
Thanks for looking into it! I just realized that `translate()` in a constructor might not work.

However, unfortunately, this change breaks this `isStaticFilter()` function, as this function assumes that the static filters are constant. After this change, it always returns a new object, so `includes()` here always returns false.

More concretely, this change breaks the fix in this PR: https://github.com/Automattic/wp-calypso/pull/71560.

So, I have another idea:

In the constructor, we only construct the keys, e.g.:

```
this.staticFilterKeys = {
  MYTHEMES: 'my-themes',
  ALL: 'all',
};
```

Then, we construct the actual filter object in `getTabFilters()`:

```
return {
	...( shouldShowMyThemesFilter && {
		MYTHEMES: {
           key: this.staticFilters.MYTHEMES,
           text: props.translate(...),
	} ),
	ALL: {
       key: this.staticFilters.ALL,
       text: props.translate(...),
	} 
	...this.subjectFilters,
};
```

What do you think?

Reply:
Hi @fushar, thank you for spotting that!!

I like your suggestion! Am I understanding correctly that we'd then change `isStaticFilter` to check by key? In that case, could it be simpler to fix the functionality of #71560 by only changing `includes` to `some`, like this?

```
isStaticFilter = ( tabFilter ) => Object.values( this.getStaticFilters() ).some( ( staticFilter ) => tabFilter.key === staticFilter.key );
```

That way, we can keep the definition of the static filters together in `getStaticFilters`, while also leaving the functionality of #71560 intact. Let me know what you think! 🙂

Reply:
I see... we can do that 😄

Reply:
Great, done! 🙂